src/fights.ts

```diff
@@ -351,6 +358,26 @@ function pygmyOptions(forceEquip: Item[] = []) {
   };
 }
 
+let _bestShadowRift: Location | null = null;
+export function bestShadowRift(): Location {
+  if (!_bestShadowRift) {
+    _bestShadowRift =
+      chooseRift({
+        canAdventure: true,
+        sortBy: (l: Location) => {
+          const drops = getMonsters(l)
+            .map((m) => Object.keys(itemDrops(m)).map((s) => toItem(s)))
+            .reduce((acc, val) => acc.concat(val), []);
```

Comment:
Why don't we want to use `flat` here? It works just fine in `autumnaton.ts`

Reply:
I guess my yarn was also a little wonky after the flat polyfill broke previously